editor/src/clj/editor/particlefx.clj

```diff
@@ -1028,36 +1050,55 @@
-                                                       (map-indexed (fn [index emitter-node]
-                                                                      [emitter-node index])))
-                                                     nodes)))
+  (output emitter-indices g/Any :cached
+          (g/fnk [^:unsafe _evaluation-context nodes]
+            ;; We use unsafe evaluation context to get child node types: these
+            ;; should never change
+            (let [basis (:basis _evaluation-context)]
+              (into {}
+                    (comp
+                      (filter #(g/node-instance? basis EmitterNode %))
+                      (map-indexed (fn [index emitter-node]
+                                     [emitter-node index])))
+                    nodes))))
 
   (output build-targets g/Any :cached produce-build-targets)
   (output scene g/Any :cached produce-scene)
   (output scene-updatable g/Any :cached produce-scene-updatable)
-  (output node-outline outline/OutlineData :cached (g/fnk [_node-id child-outlines]
-                                                     (let [[mod-outlines emitter-outlines] (let [outlines (group-by #(g/node-instance? ModifierNode (:node-id %)) child-outlines)]
-                                                                                             [(get outlines true) (get outlines false)])]
-                                                       {:node-id _node-id
-                                                        :node-outline-key "ParticleFX"
-                                                        :label "ParticleFX"
-                                                        :icon particle-fx-icon
-                                                        :children (into (outline/natural-sort emitter-outlines) (outline/natural-sort mod-outlines))
-                                                        :child-reqs [{:node-type EmitterNode
-                                                                      :tx-attach-fn (fn [self-id child-id]
-                                                                                      (attach-emitter self-id child-id true))}
-                                                                     {:node-type ModifierNode
-                                                                      :tx-attach-fn (fn [self-id child-id]
-                                                                                      (attach-modifier self-id self-id child-id true))}]})))
+  (output node-outline outline/OutlineData :cached
+          (g/fnk [^:unsafe _evaluation-context _node-id child-outlines]
+            ;; We use unsafe evaluation context to get child node types: these
+            ;; should never change
+            (let [basis (:basis _evaluation-context)
+                  outlines (group-by #(g/node-instance? basis ModifierNode (:node-id %)) child-outlines)
```

Comment:
mostly indentation, but using evaluation context here is important since it allows evaluating the `node-outline` more reliably (e.g., it works with evaluation context created in transaction, when children were also created in the same transaction).